awslogs/core.py

```diff
@@ -138,7 +139,7 @@ def consumer():
 
                 if event is do_wait:
                     if self.watch:
-                        time.sleep(1)
+                        time.sleep(self.interval)
```

Comment:
Does this solve the problem you want it to? Your PR cover mentions a 5Hz request limit, upping this number won't help that.

If you are getting [429](https://http.cat/429) codes then you at least in this case I think your rate limiting would have to be within the generator - if you have enough events and low enough latency it can hit the limit while enumerating pages.